wip-mariocao-blockweight.md

```diff
@@ -0,0 +1,282 @@
+
+- estimation of field lengths
+- do not take into consideration serialization overhead
+
+| Transaction Type |  Minimum size  |       Total size  |  Measurements  |
+| ---------------- | -------------- | ----------------- | -------------- |
+| Mint             |       40 bytes |          76 bytes |    ~  40 bytes |
+| Value Transfer   |      133 bytes |  `N*36+M*36+N*97` |    ~ 205 bytes |
+| Data Request     | 133+data bytes |   `36+N*133+data` |    ~ 400 bytes |
+| Commit           |      376 bytes |       `279+N*133` | ~ W\*412 bytes |
+| Reveal           |      149 bytes |      `149+result` | ~ W\*202 bytes |
+| Tally            | 124+data bytes |  `52+W*36+result` |    ~ 105+W\*36 |
+
+where:
+ - *N* refers to the number of inputs,
+ - *M* to the number of outputs, 
+ - *W* to the number of witnesses specified by the data request,
+ - *data* refers to length of the RADON script to be executed by witnesses,
+ - *result* refers to the length of the bytes retrieved by witnesses.
+
+
+### Specification
+
+Based on the proposal, the following parameter definitions are suggested:
+
+- A maximum block weight of 100K
+- A bounded VTT bucket of 20K
+- A bounded Data Request bucket of 80K
+- A *COMMIT_WEIGHT* of 400
+- A *REVEAL_WEIGHT* of 200
+- A *TALLY_WEIGHT* of 100
+- *beta* is equal to 1
+- *alpha* is equal to 1
+- *gamma* is equal to 10
+
+Currently the known sizes for inputs are outputs are:
+
+- INPUT_SIZE is 133 bytes (97+36)
+- OUTPUT_SIZE is 36 bytes
+
+
+#### System limits
+
+Based on the previously defined parameters, the system will be shaped in different ways:
+
+ - block chain growth
+ - value transfer throughput
+ - data request throughput
+ 
+
+**Blockchain growth**. A maximum block weight of 100K will lead to a theoretical yearly block chain growth of:
+
+```
+Blockchain growth = 365.25 days * 24 hours * 60 minutes * 60 seconds / 45 seconds * 100K bytes/block = 70 GB / year
+```
+
+**Value transfer throughput**. Three different scenarios can be considered in order to understand the value transfer throughput per block.
+
+ - Splitting transfers: the maximum UTXO splitting factor by a single transaction is 55
+     ```
+     20 KB = 1*INPUT_SIZE + M*OUTPUT_SIZE => M = 55
+     ```
+ - Joining transfers: the maximum UTXO joining by a single transaction is 147
+     ```
+     20 KB = N*INPUT_SIZE + 1*OUTPUT_SIZE => N = 147
+     ```
+ - 2-2 value transfers: the maximum number of 2-2 VTTs is 20
+     ```
+     20 KB = K * (2*INPUT_SIZE + 2*OUTPUT_SIZE) => K = 20
+     ```
+
+** Data requests throughput**. Different set scenarios have been considered in order to understand the transaction weight and its impact in the throughput. The data request size has been approximated to 400 bytes.
+
+ - Maximum committee requestable by a single Data Request is 125
```

Comment:
This is misleading
```suggestion
 - Let W be the committee requestable by a single Data Request = 125
```

Reply:
Or it is the maximum number of witnesses, because a DR with W = 126 has a weight too big to be included into a block